lib/mayaUsd/fileio/translators/translatorMayaReference.cpp

```diff
@@ -424,29 +496,88 @@ MStatus UsdMayaTranslatorMayaReference::update(const UsdPrim& prim, MObject pare
-                //
-                MString refNodeName = tempRefFn.name();
-                MString refName(refNodeName.asChar(), refNodeName.numChars() - 3);
-
-                // What reference node name is the prim expecting?
-                MString expectedRefName = refNameFromPath(parentDag);
-
-                // If found a match, reconnect the reference node's `associatedNode`
-                // attr before loading it, since the previous connection may be gone.
-                //
-                if (refName == expectedRefName) {
-                    // Reconnect the reference node's `associatedNode` attr before
-                    // loading it, since the previous connection may be gone.
-                    connectReferenceAssociatedNode(parentDag, tempRefFn);
-                    refNode = tempRefNode;
-                    break;
+        if (isMayaReference(prim)) {
+            // New behaviour for MayaReference
+            MString      expectedValue;
+            UsdAttribute attr;
+
+            if (prim.HasAttribute(MayaReferenceAttribute)) {
+                attr = prim.GetAttribute(MayaReferenceAttribute);
+                VtValue value;
+                attr.Get(&value);
+                auto resInfo = attr.GetResolveInfo();
+                // Check if attribute was directly on prim or inherited.
+                bool isAttributeOnPrim = resInfo.GetNode().GetPath() == prim.GetPath();
+                if (isAttributeOnPrim && value.GetType() == SdfValueTypeNames->String.GetType()) {
+                    expectedValue = MString(value.Get<std::string>().c_str());
+                }
+            }
+
+            // If the prim didn't have the MayaReferenceAttribute set properly used the prim info to
+            // try to match.
+            if (expectedValue.length() == 0) {
+                expectedValue = getRefNodeNameFromPrim(prim);
+            }
+
+            if (expectedValue.length() != 0) {
+                for (MItDependencyNodes refIter(MFn::kReference); !refIter.isDone();
+                     refIter.next()) {
+                    MObject      tempRefNode = refIter.item();
+                    MFnReference tempRefFn(tempRefNode);
+                    if (!tempRefFn.isFromReferencedFile()) {
+                        if (expectedValue == tempRefFn.name()) {
+                            // Reconnect the reference node's `associatedNode` attr before
+                            // loading it, since the previous connection may be gone.
+                            connectReferenceAssociatedNode(parentDag, tempRefFn);
+                            refNode = tempRefNode;
+
+                            // Update Maya Ref Node on reconnect
+                            MString uniqueRefNodeName
+                                = getUniqueRefNodeName(prim, parentDag, tempRefFn);
+                            tempRefFn.setName(uniqueRefNodeName);
```

Comment:
I still think you'll need to unlock this node before you rename, especially if something is "locking" ref nodes. If you haven't run into this yet, maybe it's possible that all the tests are creating new refs rather than updating existing ones.

Reply:
Node locking doesn't seems to be an issue. I tried modifying the "Maya Namespace" in AE and the rename is done without complaining about locking.

Reply:
ok great.